flake.nix

```diff
@@ -6,7 +6,7 @@
     rust-overlay.inputs.nixpkgs.follows = "nixpkgs";
   };
 
-  outputs = { self, nixpkgs, rust-overlay, flake-compat, ... }:
```

Comment:
`flake-compat` is used in default.nix, revert this change

Reply:
This should be a non issue though, as `default.nix` reads directly from the `flake.lock` and this is just a destructuring pattern of the function argument (`flake-compat` is not used anywhere in the `flake.nix`). I have not removed the flake input though (which AFAIK is the only necessary thing for an entry in the `flake.lock`).